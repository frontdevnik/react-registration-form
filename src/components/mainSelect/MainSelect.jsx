import React from 'react';

export default ({ name, mainText = '', options = [], register, value }) => {
  return (
    <div className="input-field col s12">
      <select name={name} ref={register}>
        <option value="" disabled selected>
          Choose your option
        </option>
        {options.map((option) => (
          <option
            key={option.value}
            value={option.value}
            selected={value === option.value && true}
          >
            {option.text}
          </option>
        ))}
      </select>
      <label>{mainText}</label>
    </div>
  );
};
