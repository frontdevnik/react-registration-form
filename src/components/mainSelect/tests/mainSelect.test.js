import React from 'react';

import MainSelect from '../MainSelect';

import { stubSelectProps } from './stubProps';

const setUp = (props) => shallow(<MainSelect {...props} />);

describe('<MainSelect />', () => {
  let component;
  let options;

  beforeEach(() => {
    component = setUp(stubSelectProps);
    options = component.find('option');
  });

  it('should render correctly', () => {
    expect(component.html()).toMatchSnapshot();
  });

  it('should main div has correct class', () => {
    expect(component.find('div').hasClass('input-field col s12')).toBeTruthy();
  });

  it('should select has name', () => {
    expect(component.find('select').props().name).toBe(stubSelectProps.name);
  });

  it('first option should render correctly', () => {
    expect(options.first().props().value).toBe('');
    expect(options.first().props().disabled).toBeDefined();
    expect(options.first().props().selected).toBeDefined();
  });

  it('all options should render correctly', () => {
    expect(options.length).toBe(stubSelectProps.options.length + 1);

    expect(options.at(1).props().value).toBe(stubSelectProps.options[0].value);
    expect(options.at(1).text()).toBe(stubSelectProps.options[0].text);

    expect(options.at(2).props().value).toBe(stubSelectProps.options[1].value);
    expect(options.at(2).text()).toBe(stubSelectProps.options[1].text);
  });

  it('should label render mainText correctly', () => {
    expect(component.find('label').text()).toBe(stubSelectProps.mainText);
  });
});
