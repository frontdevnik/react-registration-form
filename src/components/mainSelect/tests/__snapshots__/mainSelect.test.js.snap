// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainSelect /> should render correctly 1`] = `"<div class=\\"input-field col s12\\"><select name=\\"name\\"><option value=\\"\\" disabled=\\"\\" selected=\\"\\">Choose your option</option><option value=\\"value\\">text</option><option value=\\"value2\\">text2</option></select><label>Title</label></div>"`;
